lib/errors/errors.go

```diff
@@ -126,3 +126,31 @@ type ErrorResourceUpdateConflict struct {
 func (e ErrorResourceUpdateConflict) Error() string {
 	return fmt.Sprintf("update conflict: '%s'", e.Identifier)
 }
+
+// UpdateErrorIdentifier modifies the supplied error to use the new resource
+// identifier.
+func UpdateErrorIdentifier(err error, id interface{}) error {
+	if err == nil {
+		return nil
+	}
+
+	switch e := err.(type) {
```

Comment:
No we cannot.

It's a golang thang: `e := err.(type)` dereferences the interface "err" thus creating a copy of the exception struct.  So you are actually modifying a copy of the exception that is passed in.